api/libs/feature-amounts-per-planning-unit/src/feature-amounts-per-planning-units.service.ts

```diff
@@ -41,48 +41,51 @@ export class FeatureAmountsPerPlanningUnitService {
 
   public async computeMarxanAmountPerPlanningUnit(
     featureId: string,
-    scenarioId: string,
+    projectId: string,
+    geoEntityManager?: EntityManager,
   ): Promise<ComputeFeatureAmountPerPlanningUnit[]> {
     /**
      * @TODO further performance savings: limiting scans to planning_units_geom
      * by partition (we need to get the grid shape from the parent project); use
      * && operator instead of st_intersects() for bbox-based calculation of
      * intersections.
      */
+    geoEntityManager = geoEntityManager
+      ? geoEntityManager
+      : this.geoEntityManager;
+
```

Comment:
I think you could simplify this:

```suggestion
    geoEntityManager: EntityManager = this.geoEntityManager,
  ): Promise<ComputeFeatureAmountPerPlanningUnit[]> {
    /**
     * @TODO further performance savings: limiting scans to planning_units_geom
     * by partition (we need to get the grid shape from the parent project); use
     * && operator instead of st_intersects() for bbox-based calculation of
     * intersections.
     */

```